feat: Inject meta_row_number as primary key during table creation

AdaJane-SxT opened the pull request:
# Rationale for this change
We need to add the META_ROW_NUMBER column to the list of primary keys to ensure we avoid key collisions on inserts.

tlovell-sxt commented on proof-of-sql/commitment-sql/src/row_number_column.rs:
```diff
@@ -63,6 +94,37 @@ mod tests {
+            .unwrap();
+
+        assert_eq!(add_row_number_primary_key(create_table), expected);
+    }
```
This is great. I'm curious about your thoughts on...
1. Should we add a `PRIMARY KEY (META_ROW_NUMBER)` if NO primary key is defined?
2. Should we worry about any other constraints? Especially foreign keys. Maybe this is work for another PR, but we may just want to reject the other constraints available in sqlparser

AdaJane-SxT commented on proof-of-sql/commitment-sql/src/row_number_column.rs:
```diff
@@ -27,6 +27,37 @@ pub fn row_number_column_def() -> ColumnDef {
     }
 }
 
+pub fn add_row_number_primary_key(mut table: CreateTableBuilder) -> CreateTableBuilder {
```
Added

tlovell-sxt reviewed (commented):
Looks good, but some pallet-tables tests are failing.

Also NIT, the commit message and PR title don't exactly describe the full change here, since the function that is added is also called by the main logic of table creation. Something like `feat: inject meta_row_number as primary key during table creation`

AdaJane-SxT commented:
> Looks good, but some pallet-tables tests are failing.
> 
> Also NIT, the commit message and PR title don't exactly describe the full change here, since the function that is added is also called by the main logic of table creation. Something like `feat: inject meta_row_number as primary key during table creation`
Fixed those tests. Also agree with you on the commit message, so I updated it.
